packages/server/lib/util/resolve.js

```diff
@@ -21,7 +23,10 @@ module.exports = {
 
       debug('resolving typescript with options %o', options)
 
-      const resolved = resolve.sync('typescript', options)
+      // use built-in resolve when yarn v2 is detected
+      const resolved = fs.existsSync(path.join(projectRoot, '.yarn', 'cache')) ?
```

Comment:
This test is fragile and breaks as soon as someone uses the global cache or changes the cache location, if you want to detect PnP you can check if `process.versions.pnp` is defined, but you could just use `require.resolve` regardless of the runtime

```js
const resolved = require.resolve('typescript', { paths: [projectRoot] })
```
https://nodejs.org/api/modules.html#modules_require_resolve_request_options

Reply:
Though `resolve` does have PnP support so not sure why you need this

Reply:
Wonderful, thank you! I can update this quickly, and then I doubt we'd need to add tests given we are removing most of the code and reverting to the default